rpc/v2_endpoints.go

```diff
@@ -380,3 +381,178 @@ func (s *Server) GetIdentityTransfersInTickRangeV2(ctx context.Context, req *pro
 	}, nil
 
 }
+
+func (s *Server) GetQuorumTickDataV2(ctx context.Context, req *protobuff.GetQuorumTickDataRequest) (*protobuff.GetQuorumTickDataResponse, error) {
+
+	lastProcessedTick, err := s.store.GetLastProcessedTick(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "getting last processed tick: %v", err)
+	}
+	if req.TickNumber > lastProcessedTick.TickNumber {
+		st := status.Newf(codes.FailedPrecondition, "requested tick number %d is greater than last processed tick %d", req.TickNumber, lastProcessedTick.TickNumber)
+		st, err = st.WithDetails(&protobuff.LastProcessedTick{LastProcessedTick: lastProcessedTick.TickNumber})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "creating custom status")
+		}
+
+		return nil, st.Err()
+	}
+
+	processedTickIntervalsPerEpoch, err := s.store.GetProcessedTickIntervals(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "getting processed tick intervals per epoch")
+	}
+
+	wasSkipped, nextAvailableTick := wasTickSkippedByArchive(req.TickNumber, processedTickIntervalsPerEpoch)
+	if wasSkipped == true {
+		st := status.Newf(codes.OutOfRange, "provided tick number %d was skipped by the system, next available tick is %d", req.TickNumber, nextAvailableTick)
+		st, err = st.WithDetails(&protobuff.NextAvailableTick{NextTickNumber: nextAvailableTick})
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "creating custom status")
+		}
+
+		return nil, st.Err()
+	}
+
+	if req.TickNumber == lastProcessedTick.TickNumber {
+
+		tickData, err := s.store.GetQuorumTickDataV2(ctx, req.TickNumber)
+		if err != nil {
+			if errors.Is(err, store.ErrNotFound) {
+				return nil, status.Errorf(codes.NotFound, "quorum tick data not found")
+			}
+			return nil, status.Errorf(codes.Internal, "getting quorum tick data: %v", err)
+		}
+
+		res := protobuff.GetQuorumTickDataResponse{
+			QuorumTickData: &protobuff.QuorumTickData{
+				QuorumTickStructure:   tickData.QuorumTickStructure,
+				QuorumDiffPerComputor: make(map[uint32]*protobuff.QuorumDiff),
+			},
+		}
+
+		for id, diff := range tickData.QuorumDiffPerComputor {
+			res.QuorumTickData.QuorumDiffPerComputor[id] = &protobuff.QuorumDiff{
+				ExpectedNextTickTxDigestHex: diff.ExpectedNextTickTxDigestHex,
+				SignatureHex:                diff.SignatureHex,
+			}
+		}
+
+		return &res, nil
+	}
+
+	nextTick := req.TickNumber + 1
+
+	//Check if next tick was skipped
+	//TODO: check if this is accurate behaviour (if the tick gets skipped, then the data should be in the next available tick)
+	skipped, nextAvailable := wasTickSkippedByArchive(nextTick, processedTickIntervalsPerEpoch)
```

Comment:
what's thie exactly?

Reply:
In the event that the network skips a tick, the the digests won't be available for the previous tick. So we will try to get them from the next available tick.

Reply:
Removed, as the core team confirmed this is not necessary.